src/partest/scala/tools/partest/nest/AbstractRunner.scala

```diff
@@ -8,25 +8,32 @@ package partest
 package nest
 
 import utils.Properties._
-import scala.tools.nsc.Properties.{ versionMsg, propOrFalse, setProp }
+import scala.tools.nsc.Properties.{propOrFalse, setProp, versionMsg}
 import scala.collection.mutable
 import TestKinds._
 import scala.reflect.internal.util.Collections.distinctBy
+import scala.util.{ Try, Success, Failure }
+import scala.concurrent.duration.Duration
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.TimeUnit.NANOSECONDS
 
-abstract class AbstractRunner {
+class AbstractRunner(val config: RunnerSpec.Config, protected final val testSourcePath: String, val fileManager: FileManager) {
 
-  val config: RunnerSpec.Config
-
-  lazy val nestUI: NestUI = new NestUI(
+  // lazy because colorEnabled is overridden in SBTRunner
+  // TODO: integrate NestUI into AbstractRunner
```

Comment:
I was tempted to continue down this path but I should probably spend my time elsewhere at the moment.